File: GA8-220501096-AA1-EV01/NotePlusTaller.API/Models/Usuario.cs
namespace NotePlusTaller.API.Models;

public class Usuario
{
    public int IDUsuario { get; set; }
    public int IDRol { get; set; }
    public int IDPersona { get; set; }
    public string NombreUsuario { get; set; } = string.Empty;
    public string ContrasenaHash { get; set; } = string.Empty;
    public DateTime FechaCreacion { get; set; }
    public bool Estado { get; set; }

    public Rol Rol { get; set; } = null!;
    public Persona Persona { get; set; } = null!;
}

File: GA8-220501096-AA2-EV02/app/src/main/java/com/example/noteplusadmin/data/entities/Usuario.kt
package com.example.noteplusadmin.data.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "usuario",
    foreignKeys = [
        ForeignKey(
            entity = Rol::class,
            parentColumns = ["idRol"],
            childColumns = ["idRol"],
            onDelete = ForeignKey.RESTRICT
        ),
        ForeignKey(
            entity = Persona::class,
            parentColumns = ["idPersona"],
            childColumns = ["idPersona"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Usuario(
    @PrimaryKey(autoGenerate = true) val idUsuario: Int = 0,
    @ColumnInfo(index = true) val idRol: Int,
    @ColumnInfo(index = true) val idPersona: Int,
    val nombreUsuario: String,
    val contrasenaHash: String,
    val fechaCreacion: String,
    val estado: String
)
